### PR TITLE
Fix broken MSDN blogs links

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
Fixes MSDN blogs links throughout our docs and source code comments that were broken during the great MSDN migration of 2019.

Note: I am changing the contents of the _davbr Blog Archives_ folder by fixing the broken links. If we need to keep these files exactly as they were when they were copied from the blog, I can revert.

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
Good CI bot, I guess?

File: docs/coding-guidelines/clr-code-guide.md [hunk 1]
```diff
@@ -1028,14 +1028,14 @@ Here are some immediate tips for working well with the managed-debugging service
...
   - Don't assume that just because an app is under the debugger that somebody is trying to debug it.
   - Don't add additional run-time error checks when under the debugger. For example, avoid code like:  if ((IsDebuggerPresent() && (argument == null)) { throw MyException(); }
-  - Avoid massive perf changes when under the debugger. For example, don't use an interpreted stub just because you're under the debugger. We then get bugs like [my app is 100x slower when under a debugger](http://blogs.msdn.com/b/jmstall/archive/2006/01/17/pinvoke-100x-slower.aspx).
+  - Avoid massive perf changes when under the debugger. For example, don't use an interpreted stub just because you're under the debugger. We then get bugs like [my app is 100x slower when under a debugger](https://docs.microsoft.com/en-us/archive/blogs/jmstall/psa-pinvokes-may-be-100x-slower-under-the-debugger).
```
Comment (AaronRobinsonMSFT) [inline docs/coding-guidelines/clr-code-guide.md]: Please remove the locale from all URLs. (i.e. remove `en-us`). This allows the locale of the user's browser to be respected when navigating to site.

Comment (GrabYourPitchforks) [reply docs/coding-guidelines/clr-code-guide.md]: There are 343 matches for Microsoft URLs with _en-us_ in the address. Would it make sense to send a separate PR to tackle all of those at once?

Comment (AaronRobinsonMSFT) [reply docs/coding-guidelines/clr-code-guide.md]: @grabyourpitchforks sounds good. Thanks.

**Review by akoeplinger — APPROVED** [review]

**Comment by GrabYourPitchforks** [comment]
@AaronRobinsonMSFT I ran a few regexes and came up with https://github.com/GrabYourPitchforks/runtime/commit/68d024e811734ad49e7fab5ce54b700fbc1d58ae, which is an attempt to remove the hardcoded language identifiers from URLs repo-wide. I think for the most part the commit is fine, but it caught a bunch of files in the mono repo and I don't know how this would dovetail with the licensing status of those files. I'm not going to send a PR for that just yet, but wanted to give you a glimpse of how far-reaching the change would be.

**Comment by akoeplinger** [comment]
>  it caught a bunch of files in the mono repo and I don't know how this would dovetail with the licensing status of those files

Changes like that to Mono files in this repo are perfectly fine :)

**Comment by AaronRobinsonMSFT** [comment]
@GrabYourPitchforks This looks good to me.

**Pull request merged**